ui/src/project.vue

```diff
@@ -567,16 +688,47 @@ export default {
             return false;
         },
 
+        isauthor(sub) {
+            if(!Vue.config.user) return false;
+            if(Vue.config.user.sub == sub) return true;
+        },
+
         remove() {
             if(confirm("Do you really want to remove this project?")) {
                 this.$http.delete('project/'+this.selected._id)
                 .then(res=>{
                     this.selected.removed = true;
-                    this.$router.push('/project');        
+                    this.$router.push('/project');
                 });
             }
         },
 
+        submitComment() {
+            if(this.editcommentID) {
+                this.$http.patch('comment/'+this.editcommentID, {commentString: this.comment})
+                .then(res=>{
+                    console.log(res.data);
+                    /* events api will update*/
+                    // this.comments[this.editcommentIndex] = res.data;
+                    this.comment = "";
+                    this.editcommentID = null;
+                })
+            }else{
+                this.$http.post('comment/', {
+                        user_id : ""+Vue.config.user.sub,
```

Comment:
You don't need / shouldn't send user_id through the API parameter. API server should get it from decoded jwt token (`req.user.sub` on the server side)